src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,258 @@
+    }
+
+    public long getLastChecked() {
+        return lastChecked;
+    }
+
+    public void checkUpdates() {
+        UPDATING_EXCEPTIONS.clear();
+        new SwingWorker<Void, Void>() {
+            @Override
+            protected void done() {
+                pluginDisplay.refreshUI();
+            }
+
+            @Override
+            protected Void doInBackground() {
+                for (Plugin plugin : pluginHandler.LOADED_PLUGINS) {
+                    plugin.getUpdateIssues().getIssues()
+                            .removeIf(pl -> pl.getMessage().equals(I18n.get("plugins.update_issues.download_failed")));
+                    try {
+                        checkUpdate(plugin);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                }
+                lastChecked = System.currentTimeMillis();
+                return null;
+            }
+        }.execute();
+    }
+
+    private void checkUpdate(Plugin plugin) throws IOException {
+        plugin.setUpdateDefinition(findUpdate(plugin.getDefinition()));
+
+        PluginDefinition updateDef = plugin.getUpdateDefinition();
+
+        IssueHandler updateIssues = plugin.getUpdateIssues();
+        pluginHandler.testCompatibility(updateIssues, updateDef, true);
+
+        if (updateIssues.getIssues().contains(PluginHandler.NO_DOWNLOAD) ||
+                updateIssues.getIssues().contains(PluginHandler.NO_UPDATE)) {
+            plugin.setUpdateStatus(Plugin.UpdateStatus.UNKNOWN);
+            return;
+        }
+        if (plugin.getDefinition().version.compareTo(updateDef.version) >= 0) {
+            plugin.setUpdateStatus(Plugin.UpdateStatus.UP_TO_DATE);
+            return;
+        }
+
+        plugin.setUpdateStatus(updateIssues.canLoad()
+                ? Plugin.UpdateStatus.AVAILABLE
+                : Plugin.UpdateStatus.INCOMPATIBLE);
+    }
+
+    private PluginDefinition findUpdate(PluginDefinition current) throws IOException {
+        if (current.update == null) return current;
+        PluginDefinition next = pluginHandler.readPluginDefinition(current.update.openStream());
+        if (current.version.compareTo(next.version) >= 0) return current;
+        if (next.update == null || current.update.equals(next.update)) return next;
+        return findUpdate(next);
+    }
+
+    public void update(Plugin plugin) {
+        new UpdateTask(plugin).execute();
+    }
+
+    public void updateAll() {
+        new UpdateAllTask().execute();
+    }
+
+    private class UpdateAllTask extends SwingWorker<Void, Integer> {
+
+        private int progress = 0;
+        private final List<UpdateTask> updateTasks = new ArrayList<>();
+
+        UpdateAllTask() {
+            List<Plugin> availableUpdates = pluginHandler.getAvailableUpdates().collect(Collectors.toList());
+            availableUpdates.forEach(pl -> updateTasks.add(new UpdateTask(pl, this)));
```

Comment:
```suggestion
            availableUpdates = pluginHandler.getAvailableUpdates().map(pl -> new UpdateTask(pl, this).collect(Collectors.toList());
```
no need to collect to a list to later do a for-each, you can simply collect to one list directly